kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkConfigurationServiceImpl.java

```diff
@@ -336,16 +349,18 @@ private void decryptPasswordProperties(Map<String, Object> modifiedProps) throws
 
     @Override
     public synchronized ComponentConfiguration getConfiguration() throws KuraException {
-        // This method returns the network configuration properties without the current values.
-        // i.e. the ip address that should be applied to the system, but not the actual one.
+        // This method returns the network configuration properties without the current
+        // values.
+        // i.e. the ip address that should be applied to the system, but not the actual
+        // one.
         Optional<NetworkConfiguration> networkConfiguration = getNetworkConfiguration(false);
         if (!networkConfiguration.isPresent()) {
             throw new KuraRuntimeException(KuraErrorCode.CONFIGURATION_ERROR,
                     "The network component configuration cannot be retrieved");
-
         }
```

Comment:
Since the component configuration is computed from the properties and not from the networkconfiguration object, does it make sense to check if the networkconfiguration is present?